code/game/objects/items/nanoforgedkatana.dm

```diff
@@ -0,0 +1,244 @@
+			primed = TRUE
+			user.bleed(30)//about 4%
+			//desc
+			return
+	if (user == owner)
+		if (do_after(user, feeding_time, needhand = TRUE, target = user, progress = TRUE))
+			user.apply_damage(20, BRUTE, opposite_active_arm, FALSE, TRUE)
+			primed = TRUE
+			user.bleed(50)//about 9%
+			//desc
+			return
+	//anyone can become owner, but need sheathe for the dash.
+
+/obj/item/nanoforgedkatana/on_exit_storage(datum/component/storage/concrete/S)
+	var/obj/item/storage/belt/nanoforgedkatana/nanobelt = S.real_location()
+	if(istype(nanobelt))
+		playsound(nanobelt, 'sound/items/unsheath.ogg', 25, TRUE)
+		nanobelt.update_icon()
+		update_icon()
+
+/obj/item/nanoforgedkatana/on_enter_storage(datum/component/storage/concrete/S)
+	var/obj/item/storage/belt/nanoforgedkatana/nanobelt = S.real_location()
+	if(istype(nanobelt))
+		playsound(nanobelt, 'sound/items/sheath.ogg', 25, TRUE)
+		nanobelt.update_icon()
+		update_icon()
+		if(primed) // if primed, then allow dash.
+			current_sheathe = nanobelt
+			RegisterSignal(owner, COMSIG_MOB_CLICKON, .proc/predash)
+	return
+
+/obj/item/nanoforgedkatana/proc/predash(mob/user, atom/location, params)
+	SIGNAL_HANDLER
+
+	if((!length(current_sheathe.contents)) || current_sheathe.current_wearer != owner)
+		to_chat(user,"<span class='warning'>The blade needs to be in its sheathe, on its owner's body, in order to dash !")
+		UnregisterSignal(owner, COMSIG_MOB_CLICKON)
+		return // with enough fuckery, the blade can be taken out before the dash. Which is bad.
+	if(!(location in view(user)))
+		to_chat(user,"<span class='warning'>The selected dash location is out of view !")
+		return //probable if lag high, and someone clicks then moves too far before it gets processed.
+	if(!user.put_in_active_hand(src))
+		to_chat(user,"<span class='warning'>Your active hand needs to be empty for you to dash!</span>")
+		return
+	current_sheathe.update_icon()
+	update_icon()
+	UnregisterSignal(owner, COMSIG_MOB_CLICKON)
+	primed = FALSE //reset primed
+	primed_attack(location, user, src)
+	if(CanReach(location))
+		melee_attack_chain(user, location, params) //normal sword slash on final target.
+	return COMSIG_MOB_CANCEL_CLICKON
+
+/obj/item/nanoforgedkatana/proc/primed_attack(atom/target, mob/living/user, obj/item/nanoforgedkatana/sword)
+	var/turf/start = get_turf(user)
+	var/halt = FALSE
+	var/dashdir = get_dir(start, get_turf(target))
+	var/prevturf
+	var/newdir //prevturf allows to stop *before* a blocked tile, not after. newdir helps for tiles with orientation based restrictions.
+	user.setDir(dashdir)
+	// Stolen dash code from ninja
+	for(var/T in getline(start, get_turf(target)))
+		var/turf/tile = T
+		newdir = get_dir(prevturf, tile)
+		user.setDir(newdir)
+		for(var/mob/living/victim in tile)
+			if(victim != user)
+				var/Pdash = victim.stat
+				playsound(victim, 'sound/weapons/anime_slash.ogg', 10, TRUE)
+				victim.take_bodypart_damage(35) // High damage, but no armor penetration.
```

Comment:
take_bodypart_damage doesnt check armor. this IS high armor penetration, in that it completely ignores all damage modifier. use apply_damage with a blocked var equal to a melee armor check instead.

generally, you should be using either apply_damage, for most damage sources, or adjust[type]loss, for things like DOT and such that ignore armor and species modifiers

Reply:
Aaah my apologies, i thought this counted armor, yeah it's insane then jesus.
Modifying that asap, thank you !

Reply:
thats fine, damage procs arent well documented so no one really knows which ones to use. i plan to refactor apply_damage to make it a lot easier to integrate armor into the proc

Reply:
(however i have been fixated on other shit so i aint coding jack rn)